File: code/core/frontend/lib/apiv2/admin/check.ts
import type { ApiBase } from "../base";

export class AdminCheckAPI {
  base: ApiBase;
  constructor(base: ApiBase) {
    this.base = base;
  }

  bprint(bid: string) {
    this.base.get(`check/bprint/${bid}`);
  }

  plug(pid: string) {
    this.base.get(`check/plug/${pid}`);
  }

  dataGroup(gid: string) {
    this.base.get(`check/dgroup/${gid}`);
  }
}

File: code/backend/app/server/api_admin/check.go
package apiadmin

import (
	"github.com/gin-gonic/gin"
	"github.com/temphia/temphia/code/backend/xtypes/httpx"
)

func (r *ApiAdmin) checkAPI(rg *gin.RouterGroup) {
	rg.GET("/bprint/:bid", r.X(r.checkBprint))
	rg.GET("/plug/:bid", r.X(r.checkPlug))
	rg.GET("/dgroup/:bid", r.X(r.checkDataGroup))
	rg.GET("/dtable/:bid", r.X(r.checkDataTable))
}

func (r *ApiAdmin) checkBprint(ctx httpx.Request) {
	err := r.cAdmin.CheckBprint(ctx.Session, ctx.Http.Param("bid"))
	r.rutil.WriteFinal(ctx.Http, err)
}

func (r *ApiAdmin) checkPlug(ctx httpx.Request) {
	err := r.cAdmin.CheckPlug(ctx.Session, ctx.Http.Param("pid"))
	r.rutil.WriteFinal(ctx.Http, err)
}

func (r *ApiAdmin) checkDataGroup(ctx httpx.Request) {

}

func (r *ApiAdmin) checkDataTable(ctx httpx.Request) {

}
